### PR TITLE
Fixes #5373 : Add option to filter empty profiles from report

### LINKED ISSUE #5373: reporter (e.g. junit2) should skip resource packs without test execution
## Detailed Description

My inspec tests might depend on resource packs like inspec-aws, inspec-azure, inspec-* . Those resource packs themselves do not include any tests but provide resources for testing. Therefore it would be beneficial to not include those within the reporter output as those reports require post-processing.

## Context

I don't want to post-process the reporter output in order to skip the rather verbose output in regards to resource packs. By design those do not include or execute tests so it should be skipped within the reporting. If different reporters are used then my custom post-processing requires different techniques like json/xml mangling.

## Possible Implementation

a) Add a configuration option to `inspec.yml` if the tests execution of the resource pack should be included in reporter output.
b) Add a switch to `inspec ... reporter` execution to skip resource packs.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

Added option `--filter-empty-profiles` to filter the profiles from report which does not have controls in it. 
## Related Issue

Fix #5373 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/utils/run_data_filters.rb [hunk 1]
```diff
@@ -36,7 +37,15 @@ def apply_report_resize_options
...
-      # Find any inputs with :sensitive = true and replace their values with "***"
+      # Filters profiles from report which don't have controls in it.
+      def filter_empty_profiles
+        runtime_config = @config[:runtime_config]
+        if runtime_config[:filter_empty_profiles] && @run_data[:profiles].count > 1
+          @run_data[:profiles].delete_if { |p| p[:controls].empty? }
+        end
+      end
+
+      # Find any inputs with :redact_sensitive_inputsitive = true and replace their values with "***"
```
Comment (clintoncwolfe) [inline lib/inspec/utils/run_data_filters.rb]: This comment line looks to have gotten mangled... "inputsitive"

File: test/fixtures/profiles/dependencies/resource-pack/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Example InSpec Profile
```
Comment (clintoncwolfe) [inline test/fixtures/profiles/dependencies/resource-pack/README.md]: This file may be deleted.

File: test/fixtures/profiles/dependencies/uses-resource-pack/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Example InSpec Profile
```
Comment (clintoncwolfe) [inline test/fixtures/profiles/dependencies/uses-resource-pack/README.md]: This file may be deleted.

File: test/functional/inspec_exec_json_test.rb [hunk 4]
```diff
@@ -421,6 +421,30 @@
...
+        _(profiles.count).must_equal 2
+        assert_exit_code(0, run_result)
+      end
+    end
+
+    describe "with --filter-empty-profiles option" do
+      let(:run_result) { run_inspec_process("exec #{profile_path}/dependencies/uses-resource-pack --filter-empty-profiles", json: true) }
+      let(:profiles) { @json["profiles"] }
+
+      it "does not filter the empty profiles(profiles without controls)" do
```
Comment (clintoncwolfe) [inline test/functional/inspec_exec_json_test.rb]: It DOES filter the empty profiles

**Review by clintoncwolfe — COMMENTED** [review]
Just a few very minor copy-paste issues - very good work!

File: lib/inspec/utils/run_data_filters.rb (hunk 1 above)
Comment (Vasu1105) [reply lib/inspec/utils/run_data_filters.rb]: Fixed.

File: test/fixtures/profiles/dependencies/resource-pack/README.md (hunk 2 above)
Comment (Vasu1105) [reply test/fixtures/profiles/dependencies/resource-pack/README.md]: Done.

File: test/fixtures/profiles/dependencies/uses-resource-pack/README.md (hunk 3 above)
Comment (Vasu1105) [reply test/fixtures/profiles/dependencies/uses-resource-pack/README.md]: Done

File: test/functional/inspec_exec_json_test.rb (hunk 4 above)
Comment (Vasu1105) [reply test/functional/inspec_exec_json_test.rb]: Fixed

**Review by clintoncwolfe — APPROVED** [review]
Thanks!

**Review by IanMadd — APPROVED** [review]

**Pull request merged**